ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ResponseMapper.java

```diff
@@ -0,0 +1,274 @@
+            builder.responseMoreTimeNeededOption(CCDYesNoOption.valueOf(response.getMoreTimeNeeded().name()));
+        }
+
+        response.getStatementOfTruth().ifPresent(
+            statementOfTruth -> {
+                builder.responseDefendantSOTSignerName(statementOfTruth.getSignerName());
+                builder.responseDefendantSOTSignerRole(statementOfTruth.getSignerRole());
+            }
+        );
+        defendantPartyMapper.to(builder, response.getDefendant());
+
+        switch (response.getResponseType()) {
+            case FULL_DEFENCE:
+                toFullDefenceResponse(builder, (FullDefenceResponse) response);
+                break;
+            case FULL_ADMISSION:
+                toFullAdmissionResponse(builder, (FullAdmissionResponse) response);
+                break;
+            case PART_ADMISSION:
+                toPartAdmissionResponse(builder, (PartAdmissionResponse) response);
+                break;
+            default:
+                throw new MappingException("Invalid response type " + response.getResponseType());
+        }
+    }
+
+    private void toPartAdmissionResponse(CCDDefendant.CCDDefendantBuilder builder, PartAdmissionResponse response) {
+
+        builder.responseAmount(response.getAmount());
+        response.getPaymentDeclaration().ifPresent(
+            mapPaymentDeclaration(builder)
+        );
+        builder.responseDefence(response.getDefence());
+        response.getEvidence().ifPresent(mapDefendantEvidence(builder));
+        response.getTimeline().ifPresent(mapDefendantTimeline(builder));
+        response.getPaymentIntention().ifPresent(
+            paymentIntention -> builder.defendantPaymentIntention(paymentIntentionMapper.to(paymentIntention))
+        );
+
+        response.getStatementOfMeans().ifPresent(
+            statementOfMeans -> builder.statementOfMeans(statementOfMeansMapper.to(statementOfMeans))
+        );
+    }
+
+    private void toFullAdmissionResponse(CCDDefendant.CCDDefendantBuilder builder, FullAdmissionResponse response) {
+
+        builder.defendantPaymentIntention(paymentIntentionMapper.to(response.getPaymentIntention()));
+
+        response.getStatementOfMeans().ifPresent(
+            statementOfMeans -> builder.statementOfMeans(statementOfMeansMapper.to(statementOfMeans))
+        );
+    }
+
+    private void toFullDefenceResponse(
+        CCDDefendant.CCDDefendantBuilder builder,
+        FullDefenceResponse fullDefenceResponse
+    ) {
+
+        builder.responseDefenceType(
+            CCDDefenceType.valueOf(fullDefenceResponse.getDefenceType().name())
+        );
+        builder.responseDefence(fullDefenceResponse.getDefence().orElse(EMPTY));
+        fullDefenceResponse.getPaymentDeclaration().ifPresent(mapPaymentDeclaration(builder));
+        fullDefenceResponse.getEvidence().ifPresent(mapDefendantEvidence(builder));
+        fullDefenceResponse.getTimeline().ifPresent(mapDefendantTimeline(builder));
+    }
```

Comment:
Out of curiosity why we are not having paymentDeclarationMapper, EvidenceMapper and TimelineMapper instead having private methods?

Reply:
As discussed, refactoring in upcoming.